### PR TITLE
Bug/gouri-panda/2331 file reader injected in other class than normally

### LINKED ISSUE #2331: FileReader injected in other classes rather calling normally.

### PR DESCRIPTION
**Pull request by gouri-panda**
Fixes #2331

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt [hunk 1]
```diff
@@ -15,18 +15,17 @@
...
-      .bufferedReader()
-      .use(BufferedReader::readText)
-  } catch (e: IOException) {
-    "".also { e.printStackTrace() }
-  }
+fun Context.readFile(filePath: String): String = try {
+  this.assets.open(filePath)
+    .bufferedReader()
+    .use(BufferedReader::readText)
+} catch (e: Exception) {
```
Comment (abdulwd) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt:29]: Why not IOException?

Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt:29]: ok

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt [hunk 2]
```diff
@@ -15,18 +15,18 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
+@file:JvmName("FileReaderUtils")
+
 package org.kiwix.kiwixmobile.core.main
```
[the remark this answers is not in the document]
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt:20]: @s-ayush2903 Thanks for the advice!

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt [hunk 3]
```diff
@@ -15,18 +15,18 @@
...
-class FileReader {
-  fun readFile(filePath: String, context: Context): String = try {
-    context.assets.open(filePath)
-      .bufferedReader()
-      .use(BufferedReader::readText)
-  } catch (e: IOException) {
-    "".also { e.printStackTrace() }
-  }
+fun Context.readFile(filePath: String): String = try {
+  this.assets.open(filePath)
```
[the remark this answers is not in the document]
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt:27]: Sorry! Old java habit.

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java [hunk 4]
```diff
@@ -84,6 +84,7 @@
 import io.reactivex.disposables.Disposable;
 import io.reactivex.processors.BehaviorProcessor;
 import java.io.File;
+import java.io.FileReader;
```
Comment (mhutti1) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java:87]: Why is this import needed?

**Review by mhutti1 — CHANGES_REQUESTED** [review]

**Review by mhutti1 — APPROVED** [review]

**Pull request merged**